src/Microsoft.AspNet.Mvc.Core/UnitTestability/MvcControllerUnitTestHelper.cs

```diff
@@ -0,0 +1,277 @@
+				{
+					_httpContext.Request.Form = OnRequestFormCollection();
+					return _httpContext.Request;
+				}
+			}
+
+			public override HttpResponse Response
+			{
+				get
+				{
+					throw new NotImplementedException();
+				}
+			}
+
+			public override ClaimsPrincipal User
+			{
+				get
+				{
+					throw new NotImplementedException();
+				}
+
+				set
+				{
+					throw new NotImplementedException();
+				}
+			}
+
+			public override IDictionary<object, object> Items
+			{
+				get
+				{
+					throw new NotImplementedException();
+				}
+			}
+
+			public override IServiceProvider ApplicationServices
+			{
+				get
+				{
+					throw new NotImplementedException();
+				}
+
+				set
+				{
+					throw new NotImplementedException();
+				}
+			}
+
+			public override IServiceProvider RequestServices
+			{
+				get
+				{
+					return _httpContext.RequestServices;
+				}
+
+				set
+				{
+					_httpContext.RequestServices = value;
+				}
+			}
+
+			public override ISessionCollection Session
+			{
+				get
+				{
+					throw new NotImplementedException();
+				}
+			}
+
+			public override bool IsWebSocketRequest
+			{
+				get
+				{
+					throw new NotImplementedException();
+				}
+			}
+
+			public override IList<string> WebSocketRequestedProtocols
+			{
+				get
+				{
+					throw new NotImplementedException();
+				}
+			}
+
+			public override void Abort()
+			{
+				throw new NotImplementedException();
+			}
+
+			public override void Dispose()
+			{
+				throw new NotImplementedException();
+			}
+
+			public override object GetFeature(Type type)
+			{
+				throw new NotImplementedException();
+			}
+
+			public override void SetFeature(Type type, object instance)
+			{
+				throw new NotImplementedException();
+			}
+
+			public override IEnumerable<AuthenticationDescription> GetAuthenticationTypes()
+			{
+				throw new NotImplementedException();
+			}
+
+			public override IEnumerable<AuthenticationResult> Authenticate(IEnumerable<string> authenticationTypes)
+			{
+				throw new NotImplementedException();
+			}
+
+			public override Task<IEnumerable<AuthenticationResult>> AuthenticateAsync(IEnumerable<string> authenticationTypes)
+			{
+				throw new NotImplementedException();
+			}
+
+			public override Task<WebSocket> AcceptWebSocketAsync(string subProtocol)
+			{
+				throw new NotImplementedException();
+			}
+		}
+
+		private class MvcControllerTestHelpeUrlHelper : IUrlHelper
+		{
+			public Func<string, string, object, string, string, string, string>
+				OnAction
+			{
+				get;
+				set;
+			}
+
+			public string Action(string action, string controller, object values, string protocol, string host, string fragment)
+			{
+				return OnAction(action, controller, values, protocol, host, fragment);
+			}
+
+			public string Content(string contentPath)
+			{
+				throw new NotImplementedException();
+			}
+
+			public bool IsLocalUrl(string url) // Can it be a static method or the UrlHelper class or UrlHelperExtensions??
```

Comment:
You can subclass `UrlHelper` (with some tweaks) - or what's probably better is to move this to a static method and have `UrlHelper` and the testable version both call that method.